controllers/networkfence_controller.go

```diff
@@ -153,91 +187,108 @@ func (r *NetworkFenceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+}
+
+func (nf *NetworkFenceInstance) processFencingRequest() error {
 	// send FenceClusterNetwork request.
-	logger.Info("FenceClusterNetwork Request")
-	timeoutContext, cancel := context.WithTimeout(ctx, r.Timeout)
+	if nf.instance.Spec.FenceState == csiaddonsv1alpha1.Fenced {
+		nf.logger.Info("FenceClusterNetwork Request")
+	} else {
+		nf.logger.Info("UnFenceClusterNetwork Request")
+	}
+
+	request := &proto.NetworkFenceRequest{
+		Parameters:      nf.instance.Spec.Parameters,
+		SecretName:      nf.instance.Spec.Secret.Name,
+		SecretNamespace: nf.instance.Spec.Secret.Namespace,
+		Cidrs:           nf.instance.Spec.Cidrs,
+	}
+
+	if nf.instance.Spec.FenceState == csiaddonsv1alpha1.Fenced {
+		return nf.fenceClusterNetwork(request)
+	}
+
+	return nf.unfenceClusterNetwork(request)
+}
+
+func (nf *NetworkFenceInstance) fenceClusterNetwork(request *proto.NetworkFenceRequest) error {
+	timeoutContext, cancel := context.WithTimeout(nf.ctx, nf.reconciler.Timeout)
 	defer cancel()
-	_, err := controllerClient.FenceClusterNetwork(timeoutContext, &proto.NetworkFenceRequest{
-		Parameters:      nwFence.Spec.Parameters,
-		SecretName:      nwFence.Spec.Secret.Name,
-		SecretNamespace: nwFence.Spec.Secret.Namespace,
-		Cidrs:           nwFence.Spec.Cidrs,
-	})
+
+	_, err := nf.controllerClient.FenceClusterNetwork(timeoutContext, request)
+
 	if err != nil {
-		logger.Error(err, "failed to fence cluster network")
+		nf.logger.Error(err, "failed to fence cluster network")
 		return err
 	}
-	logger.Info("FenceClusterNetwork Request Succeeded")
+
+	nf.logger.Info("FenceClusterNetwork Request Succeeded")
 
 	return nil
 }
 
-// unfenceClusterNetwork sends a UnfenceClusterNetwork request and removes the finalizer on success.
-func (r *NetworkFenceReconciler) unfenceClusterNetwork(ctx context.Context, nwFence *csiaddonsv1alpha1.NetworkFence, controllerClient proto.NetworkFenceClient, logger logr.Logger) error {
-
-	timeoutContext, cancel := context.WithTimeout(ctx, r.Timeout)
+func (nf *NetworkFenceInstance) unfenceClusterNetwork(request *proto.NetworkFenceRequest) error {
+	timeoutContext, cancel := context.WithTimeout(nf.ctx, nf.reconciler.Timeout)
 	defer cancel()
-	_, err := controllerClient.UnFenceClusterNetwork(timeoutContext, &proto.NetworkFenceRequest{
-		Parameters:      nwFence.Spec.Parameters,
-		SecretName:      nwFence.Spec.Secret.Name,
-		SecretNamespace: nwFence.Spec.Secret.Namespace,
-		Cidrs:           nwFence.Spec.Cidrs,
-	})
+
+	_, err := nf.controllerClient.UnFenceClusterNetwork(timeoutContext, request)
+
 	if err != nil {
+		nf.logger.Error(err, "failed to unfence cluster network")
 		return err
 	}
-	logger.Info("UnfenceClusterNetwork Request Succeeded")
 
+	nf.logger.Info("UnFenceClusterNetwork Request Succeeded")
+
+	return nil
+}
+
+// unfenceClusterNetwork sends a UnfenceClusterNetwork request and removes the finalizer on success.
+func (nf *NetworkFenceInstance) deleteNetworkFenceResource() error {
```

Comment:
No need for this function at all? 
Call removeFinalizerFromNF directly ?

Reply:
I would suggest to add `deleteNetworkFenceResource()` or similar in the future when there is a need for it

Reply:
Or, alternatively, if it is clearer, rename `removeFinalizerFromNF` to `deleteNetworkFenceResource`?

Reply:
I don't think delete as function name makes sense,  maybe handleNetworkFenceDeletion() ?